lib/python/rose/loc_handlers/namelist.py

```diff
@@ -52,10 +53,10 @@ def parse(self, loc, conf_tree):
         loc.loc_type = loc.TYPE_BLOB
         if loc.name.endswith("(:)"):
             name = loc.name[0:-2]
-            sections = [k for k in conf_tree.node.value.keys()
+            sections = [k for k in list(conf_tree.node.value.keys())
                         if k.startswith(name)]
         else:
-            sections = [k for k in conf_tree.node.value.keys()
+            sections = [k for k in list(conf_tree.node.value.keys())
```

Comment:
*tldr; ollie ranting*

Just spotted one of my pet peeves here, using list comprehensions instead of generator expressions!

The following code is functionally the same, there is no need for either the comprehension or the `list(sections)`:

```python
else:
    sections = (k for k in list(conf_tree.node.value.keys()) if k == loc.name)
for section in sections:
    # ...
```

Reply:
What are the benefits?

Reply:
I understand your resoning, but I don't think this is that simple a change, because of the code inside the loop:
```python
            if section_value is None:
                sections.remove(section)
```

Reply:
> because of the code inside the loop:

In that case we can move the conditional check into the list comprehension.

We can ignore this though, it's not a big deal.